rust_package/brainflow/src/board_shim.rs

```diff
@@ -518,3 +513,70 @@ gen_vec_fn!(
     resistance_channels,
     "Get list of resistance channels in resulting data table for a board."
 );
+
+#[cfg(test)]
+mod tests {
+    #[cfg(test)]
+    mod functions {
+        use crate::board_shim::{get_version, get_device_name, get_eeg_names, get_board_descr};
+        use crate::{BoardIds, BrainFlowPresets};
+
+        #[test]
+        fn test_it_gets_the_version() {
+            assert_eq!("0.0.1", get_version().unwrap());
+        }
+
+        #[test]
+        fn test_get_device_name() {
+            assert_eq!("Synthetic", get_device_name(BoardIds::SyntheticBoard, BrainFlowPresets::DefaultPreset).unwrap());
+        }
+
+        #[test]
+        fn test_get_eeg_names() {
+            assert_eq!(vec!["Fz", "C3", "Cz", "C4", "Pz", "PO7", "Oz", "PO8", "F5", "F7", "F3", "F1", "F2", "F4", "F6", "F8"],
+                       get_eeg_names(BoardIds::SyntheticBoard, BrainFlowPresets::DefaultPreset).unwrap());
+
+        }
+
+        #[test]
+        fn test_get_board_descr() {
+            let expected = "{\"accel_channels\":[17,18,19],\"battery_channel\":29,\
```

Comment:
dont like this check. it probably should look like comparison with json object by keys and values not as string comparison. And for other languages `get_board_descr` returns dicts\jsons\class instances instead string, idk how to do the same in rust but should be possible

Reply:
I wrote the test reverse engineering existing functionality, using the synthetic board. It was really just enough to see that I hadnt broken existing functionality in that context (and also allowed my to prove that multiple calls caused SIGABRT). I dont really like that its bound up with a config that is coming from the C++ layers and is way out of this test. I'd probably rather test just a simple field or two here which are unlikely to change. 

Does your comment mean that you would rather we should change the contract of the function to return a json object like structure? I think its probably a good idea to do something like that, but as rust is statically typed its probably best to map it on to some types representing the the config instead of returning some sort of generic json object which is difficult to navigate. (Python etc are a lot easier/simpler for json conversions)

Reply:
>Does your comment mean that you would rather we should change the contract of the function to return a json object like structure? I think its probably a good idea to do something like that, but as rust is statically typed its probably best to map it on to some types representing the the config instead of returning some sort of generic json object which is difficult to navigate. (Python etc are a lot easier/simpler for json conversions)

yes, smth like this. In java\c# I did it using templates instead json string. https://github.com/brainflow-dev/brainflow/blob/master/java_package/brainflow/src/main/java/brainflow/examples/BandPower.java#L26 https://github.com/brainflow-dev/brainflow/blob/master/java_package/brainflow/src/main/java/brainflow/BoardDescr.java  https://github.com/brainflow-dev/brainflow/blob/master/java_package/brainflow/src/main/java/brainflow/BoardShim.java#L583

Reply:
its ok to do it in another pr, no need to change now